File: 2-Dalston/config-client/src/main/java/com/didispace/TestController.java
package com.didispace;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.cloud.context.config.annotation.RefreshScope;
import org.springframework.core.env.Environment;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RefreshScope
@RestController
public class TestController {
    @Value("${from}")
    private String from;

    @Autowired
    Environment env;

    @GetMapping("/from")
    public String from() {
        return this.from;
    }

    @GetMapping("/myhealth")
    public String health() {
        return "myhealth";
    }
    @GetMapping("/envfrom")
    public String envFrom() {
        return env.getProperty("from", "defaultValue");
    }

    @GetMapping("/cannotoverride")
    public String cannotoverride() {
        return env.getProperty("cannotovrride", "defaultValue");
    }
}

File: 2-Dalston/api-gateway-with-eureka/src/main/java/com/didispace/xueyp/controller/TestController.kt
package com.didispace.xueyp.controller

import com.didispace.xueyp.service.MyExecutorService
import com.didispace.xueyp.service.TestService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController @Autowired constructor(
    private val testService: TestService,
    private val executor: MyExecutorService
) {
    @GetMapping(value = ["bylazy"])
    fun testByLazy() {
        testService.testByLazy()
    }

    @GetMapping(value = ["observe"])
    fun testObserve() {
        testService.testObserve()
    }

    @GetMapping(value = ["value"])
    fun testValue() {
        testService.testValue()
    }

    @GetMapping(value = ["execute"])
    fun executor(@RequestParam("time") time: Int) {
        executor.submitTask(time)
    }
}